src/modules/ekf2/EKF/ekf.cpp

```diff
@@ -159,14 +138,52 @@ bool Ekf::update()
 		float filter_update_s = 1e-6f * _params.filter_update_interval_us;
 		_dt_ekf_avg = 0.99f * _dt_ekf_avg + 0.01f * math::constrain(input, 0.5f * filter_update_s, 2.f * filter_update_s);
 
+		// Filter accel for tilt initialization
+		if (_is_first_imu_sample) {
+			_accel_lpf.reset(imu_sample_delayed.delta_vel / imu_sample_delayed.delta_vel_dt);
+			_gyro_lpf.reset(imu_sample_delayed.delta_ang / imu_sample_delayed.delta_ang_dt);
+			_is_first_imu_sample = false;
+
+			_state = {};
+			initialiseTilt(_accel_lpf.getState());
+			initialiseCovariance();
+
+		} else {
+			_accel_lpf.update(imu_sample_delayed.delta_vel / imu_sample_delayed.delta_vel_dt);
+			_gyro_lpf.update(imu_sample_delayed.delta_ang / imu_sample_delayed.delta_ang_dt);
+		}
+
+		if (!_filter_initialised) {
+
+			const float accel_norm = _accel_lpf.getState().norm();
+			const float gyro_norm = _gyro_lpf.getState().norm();
+
+			if ((accel_norm > 0.8f * CONSTANTS_ONE_G)
+			    && (accel_norm < 1.2f * CONSTANTS_ONE_G)
+			    && (gyro_norm < math::radians(15.f))
+			   ) {
+				// once ready reset and init tilt from filtered accel
+				_state = {};
+				initialiseTilt(_accel_lpf.getState());
+				initialiseCovariance();
+
+				// reset the output predictor state history to match the EKF initial values
+				_output_predictor.alignOutputFilter(_state.quat_nominal, _state.vel, _gpos);
+
+				_filter_initialised = true;
+			}
+		}
+
 		updateIMUBiasInhibit(imu_sample_delayed);
```

Comment:
Why running the state and covariance prediction if the filter is not initialized?